provider/provider.go

```diff
@@ -117,20 +118,38 @@ func (p *dockerNativeProvider) Check(ctx context.Context, req *rpc.CheckRequest)
 
 	dockerContext := build.Context
 	dockerfile := build.Dockerfile
-
-	// Set docker build context
+	// Hash docker build context digest
 	contextDigest, err := hashContext(dockerContext, dockerfile)
 	if err != nil {
 		return nil, err
 	}
 
-	// add implicit resource to provider
+	// Get the system platform
+	os := runtime.GOOS
+	// Docker for Mac builds and runs for linux/amd64, not darwin
+	if os == "darwin" {
+		os = "linux"
+	}
+	arch := runtime.GOARCH
+	hostPlatform := filepath.Join(os, arch)
+	msg := fmt.Sprintf(
+		"Building your image for %s architecture.\n"+
```

Comment:
Guessing we can't make platform field required for BW-compat reasons? Or just trying to be extra helpful on defaults?

Reply:
Yes, and I would probably loosely argue that we should not make it required in any case, because this maps to implicit Docker behavior (building for host architecture/docker for mac's linux container).

Reply:
Excellent.